src/SmartComponents/SystemInventory/SystemInventoryHeader.js

```diff
@@ -48,15 +42,7 @@ const SystemInventoryHeader = ({
         'inventory:hosts:*',
         'inventory:hosts:read'
     ]);
-
-    useEffect(() => {
-        const options = { ...globalFilters(workloads, SID), ...selectedTags?.length > 0 && { tags: selectedTags } };
-        fetchInventoryTotal(options);
-        fetchInventory(options);
-        fetchInventoryStale(options);
-        fetchInventoryWarning(options);
-    }, [fetchInventoryTotal, fetchInventory, fetchInventoryStale, fetchInventoryWarning, selectedTags, workloads, SID]
-    );
+    const [isLoading, inventorySummary, inventoryWarningSummary, error] = useBatchInventoryFetch(workloads, SID, selectedTags);
```

Comment:
Can we optimise this in a way that if `hasAccess` is false, then we don't initiate any requests under this hook?

Reply:
nice note, added this now